mmv1/third_party/terraform/services/privilegedaccessmanager/data_source_privileged_access_manager_entitlement_test.go

```diff
@@ -0,0 +1,121 @@
+package privilegedaccessmanager_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccDataSourceGooglePrivilegedAccessManagerEntitlement_basic(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckPrivilegedAccessManagerEntitlementDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGooglePrivilegedAccessManagerEntitlement_basic(context),
+				Check: resource.ComposeTestCheckFunc(
+					acctest.CheckDataSourceStateMatchesResourceState("data.google_privileged_access_manager_entitlement.tfentitlement", "google_privileged_access_manager_entitlement.tfentitlement"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceGooglePrivilegedAccessManagerEntitlement_optionalProject(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckPrivilegedAccessManagerEntitlementDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGooglePrivilegedAccessManagerEntitlement_optionalProject(context),
+				Check: resource.ComposeTestCheckFunc(
+					acctest.CheckDataSourceStateMatchesResourceState("data.google_privileged_access_manager_entitlement.tfentitlement", "google_privileged_access_manager_entitlement.tfentitlement"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceGooglePrivilegedAccessManagerEntitlement_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_privileged_access_manager_entitlement" "tfentitlement" {
+	entitlement_id = "tf-test-example-entitlement%{random_suffix}"
+	location = "global"
+	max_request_duration = "43200s"
+	parent = "projects/${data.google_project.project.number}"
```

Comment:
This is causing tests to fail with:

```
Error: Reference to undeclared resource
        
          on terraform_plugin_test.tf line 6, in resource "google_privileged_access_manager_entitlement" "tfentitlement":
           6: 	parent = "projects/${data.google_project.project.number}"
```

Possibly the dependencies are order-sensitive, and the datasource needs to go at the top of the test config?